aea/aea_builder.py

```diff
@@ -393,6 +394,19 @@ def add_component(
 
         return self
 
+    def add_component_instance(self, component: Component) -> None:
+        """
+        Add already initialized component object to resources or connections.
+
+        Please, pay attention, all dependencies have to be already loaded.
+
+        :params component: Component instance already initialized.
+        """
+        if component.component_type == ComponentType.CONNECTION:
+            self._connections.append(component)
+        else:
+            self._add_component_to_resources(component)
```

Comment:
@marcofavorito I sense this is missing some consistency checks but you are more familiar with this code. Please comment.

Reply:
In any case, I think this would need to be called `load_` rather than `add_` to be in line with the rest of the builder conventions.

Reply:
I think we need to check if:
- we already have that component
- we have all the dependencies
- update the `_package_dependency_manager` object. 

basically, the same check we do on the `add_component` method.

Reply:
see https://github.com/fetchai/agents-aea/blob/v0.3.1/aea/aea_builder.py#L389-L391

Reply:
however, that means the user is responsible to provide good values in the configuration object attached to the component object. We didn't have that problem when we loaded them from the directory.

Reply:
> In any case, I think this would need to be called load_ rather than add_ to be in line with the rest of the builder conventions.

In my opinion `add_` is right, because the component is loaded outside the method. In other words, `component` is already loaded, so we just "add" it to the set of resources.

Reply:
@marcofavorito 
yes, the main reason that there is no configuration provided as it done with packages.
so it differs from the standart dependency append.
this is why it's added directly to resources.

also I assume that developer knows what he does.

Reply:
I don't think we can assume the developer knows what he does. It looks problematic to me that the checks are not applied. In particular, the dependency checks are needed to ensure the AEA can run successfully.